### PR TITLE
:test_tube: Update WeekService to correctly handle weekNums at the end of year, add tests

### PR DESCRIPTION
**Pull request by Mo-og**
tiny commit

### CODE REVIEW

**Comment by Mo-og** [comment]
|File|Coverage [23.76%]|:green_apple:|
|:-|:-:|:-:|
|[WeekService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/286d31f07a16b62c1d26a5eab3ffd09d37342178/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FWeekService.java)|100%|:green_apple:|
|[WeekController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/286d31f07a16b62c1d26a5eab3ffd09d37342178/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FWeekController.java)|20%|:green_apple:|
|[CoordinatorService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/286d31f07a16b62c1d26a5eab3ffd09d37342178/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCoordinatorService.java)|8.45%|:x:|

|Total Project Coverage|25.38%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/services/WeekService.java [hunk 1]
```diff
@@ -21,9 +20,9 @@ private UtilService() {
    * @return number of week
    */
   public static int getCurrentWeekNum() {
-    // Todo Define the current week according to ISO 8601
     var date = ZonedDateTime.now(ZoneId.of("Europe/Kiev"));
-    return date.get(ChronoField.ALIGNED_WEEK_OF_YEAR) + date.get(ChronoField.YEAR) * 100;
+    return date.get(WeekFields.ISO.weekOfWeekBasedYear())
```
Comment (olesiayakovenko) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/WeekService.java:24]: Partially duplicates code of getWeekNumByDate() method. Perhaps should use that method with today's date as an argument.

File: src/main/java/com/intellias/intellistart/interviewplanning/services/WeekService.java [hunk 2]
```diff
@@ -32,7 +31,6 @@ public static int getCurrentWeekNum() {
    * @return number of week
    */
   public static int getNextWeekNum() {
-    // Todo Define the next week according to ISO 8601
     return getCurrentWeekNum() + 1;
```
Comment (olesiayakovenko) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/WeekService.java:34]: There is an error when calculating the first week of the year when it is the next. For example, for the date 31.12.2022 the identifier of the current week should be 202252, and the next - 202301 (method getNextWeekNum() returns the result (202252 + 1) = 202253 for the next week).

**Review by olesiayakovenko — APPROVED** [review]

**Comment by Mo-og** [comment]
|File|Coverage [27.29%]|:green_apple:|
|:-|:-:|:-:|
|[WeekService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FWeekService.java)|100%|:green_apple:|
|[WebSecurityConfig.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fconfigs%2FWebSecurityConfig.java)|100%|:green_apple:|
|[UserController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FUserController.java)|86.11%|:green_apple:|
|[User.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FUser.java)|54.13%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|49.23%|:green_apple:|
|[SlotController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FSlotController.java)|35.29%|:green_apple:|
|[CustomExceptionHandler.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FCustomExceptionHandler.java)|28%|:green_apple:|
|[WeekController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FWeekController.java)|20%|:green_apple:|
|[Interviewer.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FInterviewer.java)|13.64%|:green_apple:|
|[InterviewerService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FInterviewerService.java)|10.59%|:green_apple:|
|[CoordinatorService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCoordinatorService.java)|8.45%|:x:|
|[InterviewerTimeSlot.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FInterviewerTimeSlot.java)|3.87%|:x:|
|[ApplicationErrorException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FApplicationErrorException.java)|0%|:x:|
|[InterviewerNotFoundException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FInterviewerNotFoundException.java)|0%|:x:|
|[UserNotFoundException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/5f44519944dacf3ca17f1e62ec58b84992001b93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FUserNotFoundException.java)|0%|:x:|

|Total Project Coverage|21.65%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**